src/test/java/org/jenkinsci/plugins/workflow/cps/FlowDurabilityTest.java

```diff
@@ -648,23 +654,76 @@ public void evaluate() throws Throwable {
+                    job = jenkins.createProject(WorkflowJob.class, jobName);
+                    FlowDurabilityHint hint = FlowDurabilityHint.MAX_SURVIVABILITY;
+                    TestDurabilityHintProvider provider = Jenkins.getInstance().getExtensionList(TestDurabilityHintProvider.class).get(0);
+                    provider.registerHint(job.getFullName(), hint);
+                    job.setDefinition(new CpsFlowDefinition(
+                            "echo 'first'\n" +
+                                    "def steps = [:]\n" +
+                                    "steps['1'] = {\n" +
+                                    "    echo 'do 1 stuff'\n" +
+                                    "}\n" +
+                                    "steps['2'] = {\n" +
+                                    "    echo '2a'\n" +
+                                    "    echo '2b'\n" +
+                                    "    def nested = [:]\n" +
+                                    "    nested['2-1'] = {\n" +
+                                    "        echo 'do 2-1'\n" +
+                                    "    } \n" +
+                                    "    nested['2-2'] = {\n" +
+                                    "        sleep 1\n" +
+                                    "        echo '2 section 2'\n" +
+                                    "    }\n" +
+                                    "    parallel nested\n" +
+                                    "}\n" +
+                                    "parallel steps\n" +
+                                    "echo 'final'"
+                    ));
+                }
+                story.j.buildAndAssertSuccess(job);
+                long millisDuration = job.getLastBuild().getDuration();
+
+                int time = new Random().nextInt((int) millisDuration);
+                WorkflowRun run = job.scheduleBuild2(0).getStartCondition().get();
+                Thread.sleep(time);
+                logStart[0] = JenkinsRule.getLog(run);
+                nodesOut.clear();
+                nodesOut.addAll(new DepthFirstScanner().allNodes(run.getExecution()));
+                nodesOut.sort(FlowScanningUtils.ID_ORDER_COMPARATOR);
+            }
+            });
+        story.addStep(new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                WorkflowRun run = story.j.jenkins.getItemByFullName(jobName, WorkflowJob.class).getLastBuild();
+                if (run.isBuilding()) {
+                    story.j.waitForCompletion(run);
+                    Assert.assertEquals(Result.SUCCESS, run.getResult());
+                } else {
+                    verifyCompletedCleanly(story.j.jenkins, run);
                 }
+                assertIncludesNodes(nodesOut, run);
+                story.j.assertLogContains(logStart[0], run);
+                try {
+                    FlowExecutionList.get().unregister(run.asFlowExecutionOwner());
```

Comment:
Why is this necessary? If it completed, it should be gone from FEL right? And why would you care at the end of a test anyway?